docs/Solution.rst

```diff
@@ -0,0 +1,90 @@
+    - weak: bool, i.e. whether to consider in the evaluation the weak constraints
+- *p_range(ref_value: float | None = None)*: This method is used to get the p-range of the best solution obtained in all the runs, which is the probability of obtaining a final energy lower than a certain value. The parameters are:
+    - ref_value: float | None, i.e. the reference value to calculate the p-value
+- *tts(ref_value: float | None = None, target_probability: float = 0.99)*: This method is used to get the time-to-solution of the best solution obtained in all the runs, which is the time required to obtain a solution with a certain probability. The parameters are:
+    - ref_value: float | None, i.e. the reference value to calculate the p-range
+    - target_probability: float, i.e. the target probability to calculate the time-to-solution
+-*wring_json_reports(filename: str = "report", weak: bool = False, ref_value: float | None = None, target_probability: float = 0.99, problem_features: bool = False)* : This method is used to write the reports in json format. The parameters are:
+    - filename: str, i.e. the name of the file where the report will be saved
+    - weak: bool, i.e. whether to consider in the evaluation the weak constraints
+    - ref_value: float | None, i.e. the reference value to calculate the p-range
+    - target_probability: float, i.e. the target probability to calculate the time-to-solution
+    - problem_features: bool, i.e. whether to show the problem features in the report or not
+
+
+
+Examples:
+---------
+
+.. code-block:: python
+
+    from mqt.qao.constraints import Constraints
+    from mqt.qao.variables import Variables
+    from mqt.qao.objectivefunction import ObjectiveFunction
+    from mqt.qao.problem import Problem
+    from mqt.qao.solver import Solver
+
+    variables = Variables()
+    m1 = variables.add_continuous_variables_array(
+        "M1", [1, 2], -1, 2, -1, "uniform", "logarithmic 2"
+    )
+    m2 = variables.add_continuous_variables_array(
+        "M2", [2, 1], -1, 2, -1, "uniform", "logarithmic 2"
+    )
+    objective_function = ObjectiveFunction()
+    objective_function.add_objective_function(np.matmul(m1, m2).item(0, 0))
+    constraint = Constraints()
+    constraint.add_constraint(constraint_expr, variable_precision=True)
+    problem = Problem()
+    problem.create_problem(variables, constraint, objective_function)
+    solver = Solver()
+    solution = solver.solve_simulated_annealing(
+        problem,
+        max_lambda_update=max_lambda_update,
+        lambda_update_mechanism=lambda_update,
+        lambda_strategy=lambda_strategy,
+    )
+
+    print(solution.optimal_solution_cost_functions_values())
+    print(solution.check_constraint_optimal_solution())
+    print(solution.check_constraint_all_solutions())
+    solution.show_cumulative()
+    print(solution.valid_solutions())
+    print(solution.p_range())
+    print(solution.tts())
+    solution.wring_json_reports()
```

Comment:
Given that you added examples to all these sections (which is nice), it would be nice to turn these into actual examples that are run when the documentation is being built so that you can make sure that everything works smoothly. This doesn't have to happen in this PR though. But please create an issue for reworking/extending the documentation (can be combined with the comment on the automatic API documentation).

For inspiration, you can have a look at the documentation over at the MQT Core project: https://github.com/cda-tum/mqt-core/tree/main/docs